src/Lecture2.hs

```diff
@@ -48,7 +49,10 @@ zero, you can stop calculating product and return 0 immediately.
 84
 -}
 lazyProduct :: [Int] -> Int
-lazyProduct = error "TODO"
+lazyProduct []       = 1
+lazyProduct [0, _]   = 0
```

Comment:
This pattern only works for lists of size 2. But zero can be in any list! However, you can easily fix this by using the `:` pattern 🙂 
```suggestion
lazyProduct (0 : _)   = 0
```